sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala

```diff
@@ -863,25 +863,17 @@ class InsertSuite extends DataSourceTest with SharedSparkSession {
   }
 
   test("Allow user to insert specified columns into insertable view") {
-    withSQLConf(SQLConf.USE_NULLS_FOR_MISSING_DEFAULT_COLUMN_VALUES.key -> "true") {
-      sql("INSERT OVERWRITE TABLE jsonTable SELECT a FROM jt")
-      checkAnswer(
-        sql("SELECT a, b FROM jsonTable"),
-        (1 to 10).map(i => Row(i, null))
-      )
-
-      sql("INSERT OVERWRITE TABLE jsonTable(a) SELECT a FROM jt")
-      checkAnswer(
-        sql("SELECT a, b FROM jsonTable"),
-        (1 to 10).map(i => Row(i, null))
-      )
+    sql("INSERT OVERWRITE TABLE jsonTable(a) SELECT a FROM jt")
+    checkAnswer(
+      sql("SELECT a, b FROM jsonTable"),
+      (1 to 10).map(i => Row(i, null))
```

Comment:
So the behavior of `USE_NULLS_FOR_MISSING_DEFAULT_COLUMN_VALUES` is on by default without a configuration now?

Reply:
Good question: no, because that config only applied to `INSERT INTO` commands without explicit user specified column lists. Because this command specifies explicit column `jsonTable(a)` then we add DEFAULT values for the remaining columns in the row. This behavior is the same as before where we added NULL values for the remaining columns in the row.

<img width="938" alt="image" src="https://user-images.githubusercontent.com/99207096/184430799-e76b62f8-8c2b-4cae-9fb1-73da8951ffc6.png">

Reply:
ok, so in Spark 3.3:
```
> create table foo(a int, b int);
> insert into foo (a) values (1);
Error in query: `default`.`foo` requires that the data to be inserted have the same number of columns as the target table: target table has 2 column(s) but the inserted data has 1 column(s), including 0 partition column(s) having constant value(s).
```

Seems the column default project introduces a breaking change..

Reply:
As discussed offline, this is a change but not a breaking one because if this `insert into foo (a) values (1)` command fails in Spark 3.3, but now succeeds, it is changing an error case into a successful case. This is part of the intentional behavior change of the column DEFAULT project.